### PR TITLE
Cribl Stream initial commit

### PR DESCRIPTION
**Pull request by amiracle**
### What does this PR do?

Cribl Stream Dashboard for Datadog users

### Motivation

Integrate the Cribl Stream metrics for Datadog users to visualize in their deployment.

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Comment by estherk15** [comment]
Created a card for editorial review: https://datadoghq.atlassian.net/browse/DOCS-5191

File: cribl_stream/metadata.csv
Comment (buraizu) [inline cribl_stream/metadata.csv:35]:
```suggestion
cribl.logstream.collector_cache.size ,counter,,,,Each Collector function  (default/cribl/collectors/<collector>/index.js) is loaded and initialized only once per  job, and  then cached. This metric  represents the  current size of this cache.,,,,,,,,,,,,,,,,,,
```

File: cribl_stream/metadata.csv
Comment (buraizu) [inline cribl_stream/metadata.csv:36]:
```suggestion
cribl.logstream.cluster.metrics.sender.inflight ,gauge,,,,Number of metric  packets currently being sent from a Worker or Edge Node Process to the API Process, through IPC (interprocess   communication).,,,,,,,,,,,,,,,,,,
```

File: cribl_stream/metadata.csv
Comment (buraizu) [inline cribl_stream/metadata.csv:37]:
```suggestion
cribl.logstream.backpressure.outputs ,counter,,,,Destinations experiencing backpressure, causing events to be either blocked or dropped.,,,,,,,,,,,,,,,,,,
```

File: cribl_stream/metadata.csv
Comment (buraizu) [inline cribl_stream/metadata.csv:39]:
```suggestion
cribl.logstream.pq.queue_size ,gauge,,,,Current queue size, per Destination, per Worker or Edge NodeProcess.,,,,,,,,,,,,,,,,,,
```

File: cribl_stream/metadata.csv
Comment (buraizu) [inline cribl_stream/metadata.csv:43]: Just to clarify, does `characteristic of the data` mean a [tag](https://docs.datadoghq.com/getting_started/tagging/#overview) in this context?

File: cribl_stream/metadata.csv
Comment (buraizu) [inline cribl_stream/metadata.csv:38]:
```suggestion
cribl.logstream.blocked.outputs ,counter,,,,Blocked Destinations. This metric is more restrictive than `cribl.logstream.backpressure.outputs`.,,,,,,,,,,,,,,,,,,
```

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks for the PR! It looks like some sections of the README are still under development, such as `## Support`, so this is a first-pass review per the documentation [style guide](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md).

In addition to the changes mentioned in this review, could you adjust the order of the sections to match the [README template](https://github.com/DataDog/integrations-core/blob/master/datadog_checks_dev/datadog_checks/dev/tooling/templates/integration/check/%7Bcheck_name%7D/README.md)?

Finally, just wanted to point out that the `metadata.csv` file is missing some required fields for the metrics ([reference](https://docs.datadoghq.com/developers/integrations/check_references/#metrics-metadata-file)).

Thanks again for the PR, and feel free to reach out with any questions!

File: cribl_stream/metadata.csv
Comment (amiracle) [reply cribl_stream/metadata.csv:43]: This can be a tag.

**Review by amiracle — COMMENTED** [review]
Validated all teh changes and they look good to me.

File: cribl_stream/README.md
Comment (buraizu) [inline cribl_stream/README.md:60]:
```suggestion
```

File: cribl_stream/README.md
Comment (buraizu) [inline cribl_stream/README.md:27]:
```suggestion
```

File: cribl_stream/README.md
Comment (buraizu) [inline cribl_stream/README.md:12]:
```suggestion
```

File: cribl_stream/metadata.csv
Comment (buraizu) [reply cribl_stream/metadata.csv:43]: Thanks for confirming that. Could you update the metric descriptions which currently mention `host is a characteristic of the data` to specify the tag being used? For example, is it a `<Key>:<Value>` or `<Value>` format tag? My goal is to make sure that customers and support can reference the documentation to know how to search for these hosts.

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks for applying those changes. I've just made a few additional small suggestions, mainly around formatting. I also followed up on the questions I asked around Datadog sites and a few metric descriptions. Feel free to reach out if anything is unclear or if you have any additional questions!

**Comment by anawishnoff** [comment]
Temporarily adding the product hold label until OAuth is implemented, as discussed over email. Feel free to reach out if there are any questions!

File: cribl_stream/README.md
Comment (emarsha94) [inline cribl_stream/README.md:8]: The overview section should include: 

1. What Cribl does in 1-2 sentences 
2. What will customers get from this integration (This should be clear to mutual customers but also potential new customers) 
3. Types of data being sent to Datadog

File: cribl_stream/README.md
Comment (emarsha94) [inline cribl_stream/README.md:40]: OAuth instructions will go here - Once you have tested oauth, please add the steps here

**Review by emarsha94 — CHANGES_REQUESTED** [review]
Since you are using the DD API, you will need to implement OAuth. Once you have done this, please update the PR with the new instructions. Thanks!

File: cribl_stream/CHANGELOG.md
Comment (anawishnoff) [inline cribl_stream/CHANGELOG.md:2]:
```suggestion
## 1.0.0 / 2023-06-30
* [Added] Initial Release
```

File: cribl_stream/README.md
Comment (anawishnoff) [inline cribl_stream/README.md:4]: I just want to confirm that Cribl stream is the on-prem solution that we discussed?

**Review by anawishnoff — CHANGES_REQUESTED** [review]

**Comment by anawishnoff** [comment]
Hi @amiracle, based on email discussions, I am removing the hold label from this PR. Please go ahead and address the open comments and adjust the content as necessary for the onprem offering. Once that's done, I'll do a more thorough review, as will our other reviewing teams. Thanks!

File: cribl_stream/README.md
Comment (amiracle) [reply cribl_stream/README.md:4]: Correct, this will be for our self-hosted on-prem solution to start.

File: cribl_stream/README.md
Comment (amiracle) [reply cribl_stream/README.md:40]: The self-hosted version of Cribl Stream will not be integrating with the OAuth requirement.

File: cribl_stream/README.md
Comment (amiracle) [reply cribl_stream/README.md:8]: This section seems to have the appropriate information to satisfy the requirements. If additional information is needed, please let us know and we will respond.

File: cribl_stream/CHANGELOG.md
Comment (anawishnoff) [reply cribl_stream/CHANGELOG.md:2]: The changelog addition is required, please add this in

File: cribl_stream/README.md
Comment (anawishnoff) [inline cribl_stream/README.md:10]:
```suggestion
Use the out-of-the-box dashboard to view the performance of the Stream with base metrics like events per second, bytes per second, input types, output types, and infrastructure metrics. Monitor reduction percentages by events or bytes, which is useful for improving search performance or licensing and infrastructure costs for the systems of analysis.

```
Let's add this description to the Overview tab - it will not render properly under the Data Collected tab

File: cribl_stream/README.md
Comment (anawishnoff) [inline cribl_stream/README.md:29]:
```suggestion
```

File: cribl_stream/README.md
Comment (anawishnoff) [inline cribl_stream/README.md:37]:
```suggestion
Navigate to the [API Keys page][7] under Organization Settings and create an API Key for Cribl to send data.
```

File: cribl_stream/README.md
Comment (anawishnoff) [inline cribl_stream/README.md:70]:
```suggestion
[6]: http://docs.cribl.io/logstream/sources-cribl-internal/
[7]: https://app.datadoghq.com/organization-settings/api-keys
```

File: cribl_stream/README.md
Comment (anawishnoff) [reply cribl_stream/README.md:51]: Bumping this conversation with the following suggestion:

```suggestion
6. Next, enter your _Datadog API Key_ and select your Datadog site.
```

File: cribl_stream/README.md
Comment (anawishnoff) [inline cribl_stream/README.md:60]: Would this also stop metrics from being emitted to Datadog?

File: cribl_stream/README.md
Comment (anawishnoff) [inline cribl_stream/README.md:63]: Can you add support contacts from Cribl here, and a link to any relevant documentation?

File: cribl_stream/assets/dashboards/cribl_stream_overview.json
Comment (anawishnoff) [inline cribl_stream/assets/dashboards/cribl_stream_overview.json:3]:
```suggestion
    "description": "The Cribl Stream integration for Datadog brings metrics from self-hosted Cribl Stream deployments into your monitoring dashboards. This allows operators to see the current usage of resources from the worker nodes as well as the health of the overall Cribl deployment.",
```

Comment (anawishnoff) [reply cribl_stream/assets/dashboards/cribl_stream_overview.json:3]: Fixing formatting in description

File: cribl_stream/manifest.json
Comment (anawishnoff) [inline cribl_stream/manifest.json:8]:
```suggestion
    "configuration": "README.md#Setup",
    "uninstallation": "README.md#Uninstallation",
```

File: cribl_stream/manifest.json
Comment (anawishnoff) [inline cribl_stream/manifest.json:18]: The logo will automatically be added to your tile, and we do not recommend adding it to the media carousel. Instead, could you please upload a screenshot of your dashboard? Please note that all media carousel images must follow our [guidelines](https://docs.datadoghq.com/developers/integrations/create_a_tile/?tab=buildatileontheintegrationspage#media-carousel), notably the 16:9 aspect ratio rule.

File: cribl_stream/manifest.json
Comment (anawishnoff) [inline cribl_stream/manifest.json:29]:
```suggestion
```
This will live on the Integrations page, so Marketplace is not relevant here. Please add any relevant categories from this list: https://docs.datadoghq.com/developers/integrations/check_references/#classifier-tags

File: cribl_stream/manifest.json
Comment (anawishnoff) [inline cribl_stream/manifest.json:31]:
```suggestion
      "Category::Metrics",
      "Offering::Integration"
```

**Review by anawishnoff — CHANGES_REQUESTED** [review]
Another round of comments based on recent changes, please let me know if you have questions. Thanks!

**Comment by anawishnoff** [comment]
Hi @amiracle, just wanted to ping on this PR to see if you had any questions. Thanks!

File: cribl_stream/metadata.csv
Comment (amiracle) [reply cribl_stream/metadata.csv:43]: No, actually this does not relate to a tag. This is just saying that the metric is related to a specific host sending data through Cribl Stream. The section (host is a characteristic of the data) can be removed if it's complicating the event.

File: cribl_stream/manifest.json
Comment (amiracle) [reply cribl_stream/manifest.json]: Just uploaded a screen capture of the Cribl Stream Dashboard for Datadog : cribl_stream/images/cribl_data_dog_dashboard.png .

File: cribl_stream/README.md
Comment (amiracle) [reply cribl_stream/README.md:57]: Yes, this would stop sending data to Datadog if the destination is removed from the Cribl UI.

File: cribl_stream/README.md
Comment (amiracle) [reply cribl_stream/README.md:60]: Yes, I added them to the page.

**Review by amiracle — COMMENTED** [review]
All changes have been reviewed and have been resolved.

**Comment by amiracle** [comment]
> Hi @amiracle, just wanted to ping on this PR to see if you had any questions. Thanks!

Hello @anawishnoff I've been trying to get the validation to work, but I'm stuck on the last step. It's not giving me a file or line to fix so I don't know what else to complete. Can you assist me with the current error message of asset validation failed.

File: cribl_stream/README.md
Comment (anawishnoff) [inline cribl_stream/README.md:4]: Can you please call out somewhere in the Overview section that this integration will be for on-prem customers only?

File: cribl_stream/README.md
Comment (anawishnoff) [inline cribl_stream/README.md:10]: To clarify - by reduction percentages, is this the amount of time/money/etc saved after data is optimized using Cribl? May be helpful to spell that out

File: cribl_stream/manifest.json
Comment (anawishnoff) [inline cribl_stream/manifest.json:31]: Are there any other categories that are relevant to your integration? You can find the full list [here](https://docs.datadoghq.com/developers/integrations/check_references/#classifier-tags).

**Review by anawishnoff — CHANGES_REQUESTED** [review]
A few more comments from my end - @nmuesch can you please take a look at the validation failures? Thank you!

File: cribl_stream/README.md
Comment (amiracle) [reply cribl_stream/README.md:4]: Done, I added it to the readme.md under the Setup section.

File: cribl_stream/README.md
Comment (amiracle) [reply cribl_stream/README.md:10]: Updated the text in the overview to include calling that out. "Use the Datadog dashboard for Cribl Stream to monitor reduction percentages by events or bytes. This is useful for improving search performance or licensing and infrastructure costs for the systems of analysis."

File: cribl_stream/manifest.json
Comment (amiracle) [reply cribl_stream/manifest.json:21]: rearranged the order of the image and video.

File: cribl_stream/manifest.json
Comment (amiracle) [reply cribl_stream/manifest.json:31]: Added more categories :
      "Category::AWS",
      "Category::Containers",
      "Category::Google Cloud",
      "Category::Cloud",
      "Category::Azure",
      "Category::Kubernetes",
      "Category::Log Collection",
      "Category::Security"

**Review by anawishnoff — CHANGES_REQUESTED** [review]
One last comment, otherwise looks good from my end

**Review by anawishnoff — APPROVED** [review]

**Review by alopezz — APPROVED** [review]

File: cribl_stream/README.md
Comment (buraizu) [inline cribl_stream/README.md:4]:
```suggestion
[Cribl Stream][1] helps you process machine data–logs, instrumentation data, application data, metrics, and so on–in real time, and deliver it to your analysis platform of choice. It allows you to:
```

**Review by buraizu — APPROVED** [review]
Thanks for working with us on the documentation; this looks good and I'm approving with a few final minor suggestions.
Also, just wanted to flag that `/images/cribl_dd_2.png` doesn't seem to be used in the doc. Is that intentional?

**Comment by amiracle** [comment]
Hello Ana the merge conflicts have been resolved.Cribl, Inc.Kamilo “Kam” AmirDirector, Technology ***@***.*** | Calendar Take Control and Shape Your DataOn Aug 11, 2023, at 6:58 AM, Ana Wishnoff ***@***.***> wrote:﻿
Hi @amiracle , while we're waiting on the final reviews, can you please address the merge conflicts?

**Comment by anawishnoff** [comment]
@amiracle Unfortunately it looks like there are still a few conflicts - could you resolve those? That should resolve the remaining validation errors.

**Comment by amiracle** [comment]
> @amiracle Unfortunately it looks like there are still a few conflicts - could you resolve those? That should resolve the remaining validation errors.

It looks like all the validation errors have been resolved.

**Comment by anawishnoff** [comment]
@amiracle All of the validations have passed, but there are still merge conflict on the branch

File: cribl_stream/manifest.json
Comment (anawishnoff) [inline cribl_stream/manifest.json:57]: The EULA should be a PDF file uploaded to this PR. Additionally, legal_email is not a valid field.

This doc may be helpful to address validation errors with the manifest file: https://docs.datadoghq.com/developers/integrations/check_references/#manifest-file

Comment (amiracle) [reply cribl_stream/manifest.json]: added our PDF version of our TOS v.2.04 from the site https://cribl.io/legal/terms/

**Comment by amiracle** [comment]
Disabled the synthetic test for the build.

File: cribl_stream/manifest.json
Comment (anawishnoff) [inline cribl_stream/manifest.json:63]: This section needs to be moved out of the assets object

**Comment by anawishnoff** [comment]
@alopezz or @nmuesch can you assist with the failing validations? Thanks!

File: .github/workflows/datadog-synthetics.yml
Comment (nmuesch) [inline .github/workflows/datadog-synthetics.yml:1]: This action isn't supported within this repo. Please remove this file.

**Comment by anawishnoff** [comment]
Hey @nmuesch , once you are able to approve this should be good to go. Thanks!

File: cribl_stream/manifest.json
Comment (nmuesch) [reply cribl_stream/manifest.json:50]: Please remove both this section from the manifest as well as the pdf itself. Otherwise this PR LGTM!

Comment (amiracle) [reply cribl_stream/manifest.json:50]: Removed the EULA and the `legal_terms` object.

**Review by amiracle — COMMENTED** [review]
Updated legal terms section.

**Comment by nmuesch** [comment]
Hey @amiracle It looks like your metadata.csv file is just missing the `orientation` value for its rows. Could you add that in?
Documentation on that field is here - https://docs.datadoghq.com/developers/integrations/check_references/#metrics-metadata-file Thanks!

**Review by emarsha94 — APPROVED** [review]

**Pull request merged**
